Frontend/Gmfctn/src/app/shared/components/profile-setting-modal/profile-setting-modal.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { User } from '../../models/user';
+
+@Component({
+  selector: 'app-profile-setting-modal',
+  templateUrl: './profile-setting-modal.component.html',
+  styleUrls: ['./profile-setting-modal.component.scss'],
+})
+export class ProfileSettingModalComponent implements OnInit {
+  userForm: FormGroup = new FormGroup({});
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private dialogRef: MatDialogRef<ProfileSettingModalComponent>,
+    @Inject(MAT_DIALOG_DATA) public user: User
+  ) {}
+
+  ngOnInit(): void {
+    this.userForm = this.fb.group({
+      firstName: this.fb.control(this.user.firstName, Validators.required),
+      lastName: this.fb.control(this.user.lastName, Validators.required),
+      email: this.fb.control(this.user.email, [
+        Validators.required,
+        Validators.email,
+      ]),
+      status: this.fb.control(this.user.status),
+    });
+  }
+
+  updateUser(): void {
+    if (this.userForm.valid) {
+      this.user = { ...this.userForm.value };
```

Comment:
You're changing the user object itself, which is incorrect. 
To emit the data from the dialog you should be using `this.dialogRef.close(user)` and in the parrent component `dialogRef.afterClosed().subscribe(user => {`

Reply:
I didn`t perform the endpoint for changes current user data yet. So I only changes it to show locally, only while a page is not reloaded